src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/ChainedMethodCallWrapCheck.java

```diff
@@ -0,0 +1,496 @@
+ * adhere to line length standards.
+ * </p>
+ * <p>
+ * Rationale: Long chained method calls are difficult to read and comprehend and sometimes
+ * require horizontal scrolling to read the complete method call chain. Wrapping them onto
+ * the next line/s enhances readability.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code identifierPattern} - Specify method calls to be checked. By default,
+ * identifierPattern matches nothing, so no method calls are checked. This property must be
+ * set in order to use the check.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code "^$"}.
+ * </li>
+ * <li>
+ * Property {@code maxCallsPerLine} - Specify maximum number of method calls per lines if we
+ * are wrapping the method calls. If all the method calls span over a single line then this
+ * property has no effect.
+ * Type is {@code int}.
+ * Default value is {@code 1}.
+ * </li>
+ * <li>
+ * Property {@code maxSingleLineCalls} - Specify maximum number of method calls if all chained
+ * method calls span over a single line. If chained method calls span over multiple lines then
+ * this property has no effect.
+ * Type is {@code int}.
+ * Default value is {@code 1}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check to use custom identifierPattern to validate assertions:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;ChainedMethodCallWrap&quot;&gt;
+ *  &lt;property name=&quot;identifierPattern&quot; value=&quot;^assert.*$&quot; /&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * void method() {
+ *     assertWithSomeMessage("ex").that(1).isEqualTo(1);
+ *                      // violation above, ''3' method calls on single line, max allowed is '1''
+ *     assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // violation, 'Chained method call should be wrapped'
+ *
+ *     assertWithSomeMessage("ex")
+ *         .that(1)
+ *         .isEqualTo(1);         // ok, every method call is wrapped
+ *
+ *     Truth.assertWithSomeMessage("ex").that(1).isEqualTo(1); // ok, identifierPattern is different
+ *
+ *     Truth.assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // ok, identifierPattern is different
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to use custom identifierPattern to validate Truth's assertions:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;ChainedMethodCallWrap&quot;&gt;
+ *  &lt;property name=&quot;identifierPattern&quot; value=&quot;^Truth\.assert.*$&quot; /&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * void method() {
+ *
+ *     assertWithSomeMessage("ex").that(1).isEqualTo(1); // ok, identifierPattern is different
+ *
+ *     assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // ok, identifierPattern is different
+ *
+ *     assertWithSomeMessage("ex")
+ *         .that(1)
+ *         .isEqualTo(1); // ok, every method call is wrapped, identifierPattern is also different
```

Comment:
I think some may find the comments with "also" confusing. I think that if identifier doesn't match, this should be enough explanation. @Vyom-Yadav what do you think?

Reply:
I tried to be more specific, what if the identifier pattern matches, would there be a violation?
But I have covered cases with different identifier pattern, it shouldn't be a problem removing them.

Reply:
Looks good now, thanks a lot. resolved.